node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+        let cw_service_fee_balance = 111_222_333_444;
+        let largest_exceeding_balance_recently_qualified = 3_555_666;
+        let now = SystemTime::now();
+
+        subject.initialize_inner(
+            cw_service_fee_balance,
+            required_adjustment,
+            largest_exceeding_balance_recently_qualified,
+            now,
+        );
+
+        assert_eq!(subject.inner.now(), now);
+        assert_eq!(
+            subject.inner.transaction_fee_count_limit_opt(),
+            expected_tx_fee_limit_opt_result
+        );
+        assert_eq!(
+            subject.inner.original_cw_service_fee_balance_minor(),
+            cw_service_fee_balance
+        );
+        assert_eq!(
+            subject.inner.unallocated_cw_service_fee_balance_minor(),
+            cw_service_fee_balance
+        );
+        assert_eq!(
+            subject.inner.largest_exceeding_balance_recently_qualified(),
+            largest_exceeding_balance_recently_qualified
+        )
+    }
+
+    #[test]
+    fn initialize_inner_processes_works() {
+        test_initialize_inner_works(Adjustment::ByServiceFee, None);
+        test_initialize_inner_works(
+            Adjustment::TransactionFeeInPriority {
+                affordable_transaction_count: 5,
+            },
+            Some(5),
+        );
+    }
+
+    #[test]
+    fn search_for_indispensable_adjustment_happy_path() {
+        init_test_logging();
+        let test_name = "search_for_indispensable_adjustment_gives_negative_answer";
+        let mut subject = PaymentAdjusterReal::new();
+        subject.logger = Logger::new(test_name);
+        // Service fee balance > payments
+        let input_1 = make_input_for_initial_check_tests(
+            Some(TestConfigForServiceFeeBalances {
+                account_balances: Either::Right(vec![
+                    gwei_to_wei::<u128, u64>(85),
+                    gwei_to_wei::<u128, u64>(15) - 1,
+                ]),
+                cw_balance_minor: gwei_to_wei(100_u64),
+            }),
+            None,
+        );
+        // Service fee balance == payments
+        let input_2 = make_input_for_initial_check_tests(
+            Some(TestConfigForServiceFeeBalances {
+                account_balances: Either::Left(vec![85, 15]),
+                cw_balance_minor: gwei_to_wei(100_u64),
+            }),
+            None,
+        );
+        // Transaction fee balance > payments
+        let input_3 = make_input_for_initial_check_tests(
+            None,
+            Some(TestConfigForTransactionFees {
+                agreed_transaction_fee_per_computed_unit_major: 100,
+                number_of_accounts: 6,
+                estimated_transaction_fee_units_per_transaction: 53_000,
+                cw_transaction_fee_balance_major: {
+                    let base_value = 100 * 6 * 53_000;
+                    let exact_equality = TRANSACTION_FEE_MARGIN.add_percent_to(base_value);
+                    exact_equality + 1
```

Comment:
Some name that tells that this variable's accuracy is related to the test, it'll make it better.